src/components/vacation-requests-table/vacation-requests-table-toolbar/vacation-requests-table-toolbar.tsx

```diff
@@ -123,7 +123,8 @@ const TableToolbar = ({
           >
             <ToolbarDeleteButton setConfirmationHandlerOpen={setConfirmationHandlerOpen} />
           </ToolbarGridItem>
-          {selectedRowIds?.length === 1 && (
+          {selectedRowIds?.length === 1 &&
+          rows.find((request: VacationsDataGridRow) => request.id === selectedRowIds[0])?.status !== VacationRequestStatuses.APPROVED &&
```

Comment:
I would say this is a little more inline logic for a conditional render than is ideal, its not especially clear/ readable. I would suggest creating some kind of disableEdit variable which takes the logic e.g.

const disableEdit = selectedRowIds?.length === 1 && rows.find((request: VacationsDataGridRow) => request.id === selectedRowIds[0])?.status !== VacationRequestStatuses.PENDING

You could then pass the disableEdit to the FormToggleButton and you would not need to have a different component for each condition rather a single condition with only the disabled prop changing